Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/UserMetadataServiceImpl.java

```diff
@@ -281,65 +280,94 @@ public void deleteUserMetadata(Integer usermetadataId, String username) throws U
     }
 
     @Override
-    public Set<Usermetadata> getUserMetadata(String username, List<Integer> logIds,
-                                             UserMetadataTypeEnum userMetadataTypeEnum) throws UserNotFoundException {
-
-        if (null == logIds || logIds.size() == 0) {
-            return null;
-        }
+    public Set<Usermetadata> getUserMetadataByUser(String username, UserMetadataTypeEnum userMetadataTypeEnum) throws UserNotFoundException {
 
         User user = userSrv.findUserByLogin(username);
         assert user != null;
 
         // Get all the user metadata that can be accessed by groups that contain specified user
-        Set<GroupUsermetadata> groupUsermetadataSet = new HashSet<>();
+        Set<Usermetadata> umSet = new HashSet<>();
+
         for (Group group : user.getGroups()) {
-            groupUsermetadataSet.addAll(groupUsermetadataRepo.findByGroup(group));
-        }
-        Set<Usermetadata> usermetadataList1 = new HashSet<>();
-        for (GroupUsermetadata groupUsermetadata : groupUsermetadataSet) {
-            Usermetadata u = groupUsermetadata.getUsermetadata();
-            if (u.getUsermetadataType().getId().equals(userMetadataTypeEnum.getUserMetadataTypeId()) && u.getIsValid()) {
-                usermetadataList1.add(u);
+            List<GroupUsermetadata> guList = groupUsermetadataRepo.findByGroup(group);
+
+            for (GroupUsermetadata gu : guList) {
+                Usermetadata u = gu.getUsermetadata();
+                if (u.getUsermetadataType().getId().equals(userMetadataTypeEnum.getUserMetadataTypeId()) && u.getIsValid()) {
+                    umSet.add(u);
+                }
             }
         }
+        return umSet;
+    }
+
+    @Override
+    public Set<Usermetadata> getUserMetadataByLogs(List<Integer> logIds, UserMetadataTypeEnum userMetadataTypeEnum) {
+
+        if (null == logIds || logIds.size() == 0) {
+            return null;
+        }
 
```

Comment:
Return an empty set would help avoid NPE at the calling side and make code less "if else" block.